File: benchmarks/systems/alloc_1m.go
package main

import (
	"fmt"
	"time"
)

type Obj struct {
	ID   int
	Data [10]int
}

func main() {
	start := time.Now()
	objs := make([]Obj, 1000000)
	for i := range objs {
		objs[i].ID = i
	}
	objs = nil
	_ = objs
	fmt.Println(time.Since(start).Microseconds())
}

File: benchmarks/systems/alloc_1m.nim
import std/times
type Obj = object
  id: int
  data: array[10, int]
let start = cpuTime()
var objs = newSeq[Obj](1000000)
for i in 0..<1000000:
  objs[i].id = i
objs.setLen(0)
echo int((cpuTime() - start) * 1_000_000)
